src/main/java/org/opensearch/knn/plugin/script/KNNScoringSpaceUtil.java

```diff
@@ -137,4 +146,43 @@ public static float getVectorMagnitudeSquared(float[] inputVector) {
         }
         return normInputVector;
     }
+
+    /**
+     * Get the expected dimensions from a specified knn vector field type.
+     *
+     * If the field is model-based, get dimensions from model metadata.
+     * @param knnVectorFieldType knn vector field type
+     * @return expected dimensions
+     */
+    public static int getExpectedDimensions(KNNVectorFieldMapper.KNNVectorFieldType knnVectorFieldType) {
```

Comment:
Can we move this to KNNVectorFieldMapperUtil?

Reply:
Sure